diff -ru orig/duplicity-0.6.21/duplicity/commandline.py new/duplicity-0.6.21/duplicity/commandline.py
--- orig/duplicity-0.6.21/duplicity/commandline.py	2013-01-23 09:35:56.000000000 -0500
+++ new/duplicity-0.6.21/duplicity/commandline.py	2013-04-17 11:19:02.000000000 -0400
@@ -245,6 +245,9 @@
     # source directory doesn't match previous backup source directory.
     parser.add_option("--allow-source-mismatch", action="store_true")
 
+    # If this is true generate the alternate filenames.
+    parser.add_option("--alternate-filenames", action="store_true")
+
     # Set to the path of the archive directory (the directory which
     # contains the signatures and manifests of the relevent backup
     # collection), and for checkpoint state between volumes.
diff -ru orig/duplicity-0.6.21/duplicity/file_naming.py new/duplicity-0.6.21/duplicity/file_naming.py
--- orig/duplicity-0.6.21/duplicity/file_naming.py	2013-01-23 09:35:56.000000000 -0500
+++ new/duplicity-0.6.21/duplicity/file_naming.py	2013-04-17 11:46:58.000000000 -0400
@@ -69,13 +69,13 @@
                                "(?P<partial>(\\.p))?"
                                "($|\\.)")
 
-    full_manifest_re = re.compile("^" + globals.file_prefix + "duplicity-full"
+    full_manifest_re = re.compile("^" + globals.file_prefix + "duplicity-full(-manifest|)"
                               "\\.(?P<time>.*?)"
                               "\\.manifest"
                               "(?P<partial>(\\.part))?"
                               "($|\\.)")
 
-    full_manifest_re_short = re.compile("^" + globals.file_prefix + "df"
+    full_manifest_re_short = re.compile("^" + globals.file_prefix + "df(-m|)"
                                     "\\.(?P<time>[0-9a-z]+?)"
                                     "\\.m"
                                     "(?P<partial>(\\.p))?"
@@ -95,7 +95,7 @@
                               "\\.dt"
                               "($|\\.)")
 
-    inc_manifest_re = re.compile("^" + globals.file_prefix + "duplicity-inc"
+    inc_manifest_re = re.compile("^" + globals.file_prefix + "duplicity-inc(-manifest|)"
                              "\\.(?P<start_time>.*?)"
                              "\\.to"
                              "\\.(?P<end_time>.*?)"
@@ -103,7 +103,7 @@
                              "(?P<partial>(\\.part))?"
                              "(\\.|$)")
 
-    inc_manifest_re_short = re.compile("^" + globals.file_prefix + "di"
+    inc_manifest_re_short = re.compile("^" + globals.file_prefix + "di(-m|)"
                                    "\\.(?P<start_time>[0-9a-z]+?)"
                                    "\\.(?P<end_time>[0-9a-z]+?)"
                                    "\\.m"
@@ -122,7 +122,7 @@
                                "(?P<partial>(\\.p))?"
                                "(\\.|$)")
 
-    new_sig_re = re.compile("^" + globals.file_prefix + "duplicity-new-signatures"
+    new_sig_re = re.compile("^" + globals.file_prefix + "duplicity-(new|inc)-signatures"
                         "\\.(?P<start_time>.*?)"
                         "\\.to"
                         "\\.(?P<end_time>.*?)"
@@ -130,7 +130,7 @@
                         "(?P<partial>(\\.part))?"
                         "(\\.|$)")
 
-    new_sig_re_short = re.compile("^" + globals.file_prefix + "dns"
+    new_sig_re_short = re.compile("^" + globals.file_prefix + "(dns|dis)"
                               "\\.(?P<start_time>[0-9a-z]+?)"
                               "\\.(?P<end_time>[0-9a-z]+?)"
                               "\\.st"
@@ -225,12 +225,14 @@
                         (dup_time.curtimestr, part_string, suffix))
         elif type == "new-sig":
             if globals.short_filenames:
-                return ("dns.%s.%s.st%s%s" %
-                        (to_base36(dup_time.prevtime), to_base36(dup_time.curtime),
+                return ("d%ss.%s.%s.st%s%s" %
+                        ("i" if globals.alternate_filenames else "n",
+                         to_base36(dup_time.prevtime), to_base36(dup_time.curtime),
                          part_string, suffix))
             else:
-                return ("duplicity-new-signatures.%s.to.%s.sigtar%s%s" %
-                        (dup_time.prevtimestr, dup_time.curtimestr,
+                return ("duplicity-%s-signatures.%s.to.%s.sigtar%s%s" %
+                        ("inc" if globals.alternate_filenames else "new",
+                         dup_time.prevtimestr, dup_time.curtimestr,
                          part_string, suffix))
     else:
         assert volume_number or manifest
@@ -247,20 +249,28 @@
                 vol_string = "manifest"
         if type == "full":
             if globals.short_filenames:
-                return ("df.%s.%s%s%s" % (to_base36(dup_time.curtime),
-                                          vol_string, part_string, suffix))
+                return ("df%s.%s.%s%s%s" % ("-m" if (globals.alternate_filenames
+                                                     and manifest) else "",
+                                            to_base36(dup_time.curtime),
+                                            vol_string, part_string, suffix))
             else:
-                return ("duplicity-full.%s.%s%s%s" % (dup_time.curtimestr,
-                                                      vol_string, part_string, suffix))
+                return ("duplicity-full%s.%s.%s%s%s" % ("-manifest" if (globals.alternate_filenames
+                                                                        and manifest) else "",
+                                                        dup_time.curtimestr,
+                                                        vol_string, part_string, suffix))
         elif type == "inc":
             if globals.short_filenames:
-                return ("di.%s.%s.%s%s%s" % (to_base36(dup_time.prevtime),
-                                             to_base36(dup_time.curtime),
-                                             vol_string, part_string, suffix))
+                return ("di%s.%s.%s.%s%s%s" % ("-m" if (globals.alternate_filenames
+                                                     and manifest) else "",
+                                               to_base36(dup_time.prevtime),
+                                               to_base36(dup_time.curtime),
+                                               vol_string, part_string, suffix))
             else:
-                return ("duplicity-inc.%s.to.%s.%s%s%s" % (dup_time.prevtimestr,
-                                                           dup_time.curtimestr,
-                                                           vol_string, part_string, suffix))
+                return ("duplicity-inc%s.%s.to.%s.%s%s%s" % ("-manifest" if (globals.alternate_filenames
+                                                                        and manifest) else "",
+                                                             dup_time.prevtimestr,
+                                                             dup_time.curtimestr,
+                                                             vol_string, part_string, suffix))
         else:
             assert 0
 
diff -ru orig/duplicity-0.6.21/duplicity/globals.py new/duplicity-0.6.21/duplicity/globals.py
--- orig/duplicity-0.6.21/duplicity/globals.py	2013-01-23 09:35:57.000000000 -0500
+++ new/duplicity-0.6.21/duplicity/globals.py	2013-04-17 11:19:00.000000000 -0400
@@ -94,6 +94,9 @@
 # source directory doesn't match previous backup source directory.
 allow_source_mismatch = None
 
+# If this is true generate the alternate filenames.
+alternate_filenames = None
+
 # If set, abort if cannot do an incremental backup.  Otherwise if
 # signatures not found, default to full.
 incremental = None
